packages/forms/src/components/form.gts

```diff
@@ -1,26 +1,74 @@
 import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
 import { action } from '@ember/object';
 import { on } from '@ember/modifier';
 import { dataFrom } from 'form-data-utils';
 
 type FormResultData = ReturnType<typeof dataFrom>;
 
+/**
+ * The context yielded to the default block of the `Form` component.
+ */
+interface FormContext {
+  /** Whether the form is currently submitting. */
+  isLoading: boolean;
+}
+
 interface FormSignature {
   Element: HTMLFormElement;
   Args: {
+    /**
+     * Optional callback invoked on input changes within the form.
+     */
     onChange?: (data: FormResultData, event: Event) => void;
```

Comment:
We probably should change the return type to be `Promise<void> | void`.

When we get to validation, we could also allow it to return a list of errors from backend validations, but we can discuss that when we reach that part.